### PR TITLE
add english userguide.

### PR DESCRIPTION
**Pull request by YuJuncen**
I translated the user guide at `docs/cn` into English.

### CODE REVIEW

File: docs/en/user-guide.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+| -P or --port | TCP/IP port to connect to. (default 4000) |
+| -u or --user | Username with privileges to run the dump. (default "root") |
+
+More detailed usage can be find by the flag `-h` or `--help`.
+
+## Mydumper Reference
+
+[Mydumper usage](https://github.com/maxbube/mydumper/blob/master/docs/mydumper_usage.rst)
+
+[TiDB Mydumper Manual (Chinese)](https://pingcap.com/docs-cn/stable/reference/tools/mydumper/)
```
Comment (3pointer) [inline docs/en/user-guide.md:38]:
```suggestion
[TiDB Mydumper Manual](https://pingcap.com/docs/stable/reference/tools/mydumper/)
```

File: docs/en/user-guide.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Dumpling manual
+
+[Dumpling](https://github.com/pingcap/dumpling) is a tool that supports export MySQL/TiDB data as SQL text or CSV file.
```
Comment (kennytm) [inline docs/en/user-guide.md:3]:
```suggestion
[Dumpling](https://github.com/pingcap/dumpling) is a tool that supports exporting data from MySQL/TiDB as SQL or CSV files.
```

File: docs/en/user-guide.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+It's designed to replace [Mydumper](https://github.com/pingcap/mydumper), so you can learn its basic usage from Mydumper. Of course, we didn't make a replica of Mydumper, so there are difference between them.
```
Comment (kennytm) [inline docs/en/user-guide.md:5]: This paragraph sounds not so professional. Perhaps just copy the statement from the README.

File: docs/en/user-guide.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+The following table lists main parameters of dumpling.
+
+
+| Parameter |     |
+| --------| --- |
+| -B or --database | Export to specifcated database. |
```
Comment (kennytm) [inline docs/en/user-guide.md:12]:
```suggestion
| -B or --database | Export the specified database. |
```

File: docs/en/user-guide.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+| -H or --host | Address of node to connect. (default 127.0.0.1) |
```
Comment (kennytm) [inline docs/en/user-guide.md:13]:
```suggestion
| -H or --host | Host to connect to. (default 127.0.0.1) |
```

File: docs/en/user-guide.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+# Dumpling manual
+
+[Dumpling](https://github.com/pingcap/dumpling) is a tool that supports export MySQL/TiDB data as SQL text or CSV file.
+
+It's designed to replace [Mydumper](https://github.com/pingcap/mydumper), so you can learn its basic usage from Mydumper. Of course, we didn't make a replica of Mydumper, so there are difference between them.
+
+The following table lists main parameters of dumpling.
```
Comment (kennytm) [inline docs/en/user-guide.md:7]: Better just copy the output from `./dumpling -h`.

File: docs/en/user-guide.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+
+| Parameter |     |
+| --------| --- |
+| -B or --database | Export to specifcated database. |
+| -H or --host | Address of node to connect. (default 127.0.0.1) |
+| -t or --threads | Thread number for concurrence backup. |
```
Comment (kennytm) [inline docs/en/user-guide.md:14]:
```suggestion
| -t or --threads | Number of threads for concurrent backup. |
```

File: docs/en/user-guide.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+| -r or --rows | Split table to rows of data. When working with big tables, normally we generate many files concurrencly. |
```
Comment (kennytm) [inline docs/en/user-guide.md:15]:
```suggestion
| -r or --rows | Split table into multiple files by number of rows. This allows Dumpling to generate multiple files concurrently. |
```

File: docs/en/user-guide.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+| --loglevel | Log level. (debug, info, warn, error, dpanic, panic, fatal, default "info") |
+| -d or --no-data | Don't export data, for schema-only case. |
+| --no-header | Export table csv without header. |
+| -W or --no-views | Don't export views. (default true) |
+| -m or --no-schemas | Don't export schema, export data only. |
+| -s or --statement-size | Control the size of Insert Statement, unit is byte. |
+| -F or --filesize | The approximate size of output file, unit is byte. |
+| --filetype| The type of export file. (sql/csv, default "sql")           |
+| -o or --output | Output directory. (We have a default value based on time.) |
+| --consistency | Consistency level during dumping: (deault `auto`)<br>`flush`: FTWRL before dump <br>`snapshot`: specify dump timestamp by tso <br>`lock`: execute lock tables read for all tables need to be locked <br>`none`: dump without locking, cannot guarantee consistency <br>`auto`: flush on MySQL, snapshot on TiDB |
```
Comment (kennytm) [inline docs/en/user-guide.md:25]:
```suggestion
| --consistency | Which consistency control to use (default `auto`):<br>`flush`: Use FTWRL (flush tables with read lock)<br>`snapshot`: use a snapshot at given timestamp<br>`lock`: execute lock tables read for all tables need to be locked <br>`none`: dump without locking, cannot guarantee consistency <br>`auto`: `flush` on MySQL, `snapshot` on TiDB |
```

File: docs/en/user-guide.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+More detailed usage can be find by the flag `-h` or `--help`.
+
+## Mydumper Reference
+
+[Mydumper usage](https://github.com/maxbube/mydumper/blob/master/docs/mydumper_usage.rst)
+
+[TiDB Mydumper Manual](https://pingcap.com/docs/stable/reference/tools/mydumper/)
+
+## Dumpling Downloading
```
Comment (kennytm) [inline docs/en/user-guide.md:40]:
```suggestion
## Download
```

File: docs/en/user-guide.md (hunk 6 above)
Comment (YuJuncen) [reply docs/en/user-guide.md:7]: Sounds good, do we need to change Chinese document at the same time?

Comment (YuJuncen) [reply docs/en/user-guide.md:7]: Some of docs are copied from CLI output, and the description at CLI help message seems isn't clear as this doc, maybe leave this doc as is would also be good.

Comment (kennytm) [reply docs/en/user-guide.md:7]: > Sounds good, do we need to change Chinese document at the same time?

Well the `-h` output is English only 😐. You could change the Chinese doc to fit the meaning English but I think not much needs to be changed.

Comment (YuJuncen) [reply docs/en/user-guide.md:7]: Some of flags that `-h` prints sees isn't present at this doc(e.g. `--csv-null-value`), and this doc has mentioned `-h` flag at last.

I think it's better to keep consistency with Chinese doc version... Making a replica of existing resources seems a little strange... 🤔

File: docs/en/user-guide.md [hunk 11]
```diff
@@ -1,8 +1,8 @@
 # Dumpling manual
 
-[Dumpling](https://github.com/pingcap/dumpling) is a tool that supports exporting data from MySQL/TiDB as SQL or CSV files.
+**Dumpling** is a tool and a Go library for creating SQL dump (CSV/SQL format) from a MySQL-compatible database.
 
-It's designed to replace [Mydumper](https://github.com/pingcap/mydumper), so you can learn its basic usage from Mydumper. Of course, we didn't make a replica of Mydumper, so there are difference between them.
+It is intended to replace `mysqldump` and `mydumper` when targeting TiDB, as a result, its basic usage is similar to Mydumper. Of course, we didn't make a replica of Mydumper, so there are difference between them.
```
Comment (kennytm) [inline docs/en/user-guide.md:5]: When I say "unprofessional" I mean that sentence starting with "Of course,".

Comment (YuJuncen) [reply docs/en/user-guide.md]: I got it. 🙈

File: docs/en/user-guide.md (hunk 6 above)
Comment (kennytm) [reply docs/en/user-guide.md:8]: The documentations should be consistent with the actual behavior of the program, not with each other 🤔.

File: docs/en/user-guide.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
+# Dumpling manual
+
+**Dumpling** is a tool and a Go library for creating SQL dump (CSV/SQL format) from a MySQL-compatible database.
+
+It is intended to replace `mysqldump` and `mydumper` when targeting TiDB, as a result, its basic usage is similar to Mydumper. 
+See detail at the following table.
```
Comment (kennytm) [inline docs/en/user-guide.md:6]:
```suggestion
```

File: docs/en/user-guide.md [hunk 13]
```diff
@@ -3,29 +3,28 @@
...
 | -t or --threads | Number of threads for concurrent backup. |
-| -r or --rows | Split table into multiple files by number of rows. This allows Dumpling to generate multiple files concurrently. |
+| -r or --rows | Split table into chunks of this many rows. (default unlimited) |
 | --loglevel | Log level. (debug, info, warn, error, dpanic, panic, fatal, default "info") |
-| -d or --no-data | Don't export data, for schema-only case. |
-| --no-header | Export table csv without header. |
-| -W or --no-views | Don't export views. (default true) |
-| -m or --no-schemas | Don't export schema, export data only. |
+| -d or --no-data | Don't dump data, for schema-only case. |
+| --no-header | DUmp table csv without header. |
```
Comment (kennytm) [inline docs/en/user-guide.md:18]:
```suggestion
| --no-header | Dump table CSV without header. |
```

😅

**Comment by GregoryIan** [comment]
LGTM

File: docs/en/user-guide.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
+# Dumpling manual
+
+**Dumpling** is a tool and a Go library for creating SQL dump (CSV/SQL format) from a MySQL-compatible database.
+
+It is intended to replace `mysqldump` and `mydumper` when targeting TiDB, as a result, its basic usage is similar to Mydumper.
```
Comment (ran-huang) [inline docs/en/user-guide.md:5]:
```suggestion
It is intended to replace `mysqldump` and `mydumper` when targeting TiDB; as a result, its basic usage is similar to that of Mydumper. 
```

File: docs/en/user-guide.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
+# Dumpling manual
+
+**Dumpling** is a tool and a Go library for creating SQL dump (CSV/SQL format) from a MySQL-compatible database.
+
+It is intended to replace `mysqldump` and `mydumper` when targeting TiDB, as a result, its basic usage is similar to Mydumper. 
+
+The following table lists main parameters of dumpling.
```
Comment (ran-huang) [inline docs/en/user-guide.md:7]:
```suggestion
The following table lists the major parameters of Dumpling.
```

File: docs/en/user-guide.md [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+
+
+| Parameter |     |
```
Comment (ran-huang) [inline docs/en/user-guide.md:10]:
```suggestion
| Parameter | Description |
```

File: docs/en/user-guide.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+| --------| --- |
+| -B or --database | Dump the specified database. |
+| -H or --host | Host to connect to. (default 127.0.0.1) |
```
Comment (ran-huang) [inline docs/en/user-guide.md:13]:
```suggestion
| -H or --host | Host to connect to. (default: `127.0.0.1`) |
```

File: docs/en/user-guide.md [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+| -t or --threads | Number of threads for concurrent backup. |
+| -r or --rows | Split table into chunks of this many rows. (default unlimited) |
```
Comment (ran-huang) [inline docs/en/user-guide.md:15]: This seems inconsistent with the Chinese version.
```suggestion
| -r or --rows | Number of rows the table is split into. (default: unlimited) |
```

File: docs/en/user-guide.md [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+| --loglevel | Log level. (debug, info, warn, error, dpanic, panic, fatal, default "info") |
```
Comment (ran-huang) [inline docs/en/user-guide.md:16]:
```suggestion
| --loglevel | Log level. {debug, info, warn, error, dpanic, panic, fatal}. (default: `info`) |
```

File: docs/en/user-guide.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+| -d or --no-data | Don't dump data, for schema-only case. |
+| --no-header | Dump table CSV without header. |
+| -W or --no-views | Don't dump views. (default true) |
```
Comment (ran-huang) [inline docs/en/user-guide.md:19]:
```suggestion
| -W or --no-views | Don't dump views. (default: `true`) |
```

File: docs/en/user-guide.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+| -m or --no-schemas | Don't dump schemas, dump data only. |
+| -s or --statement-size | Control the size of Insert Statement, unit is byte. |
```
Comment (ran-huang) [inline docs/en/user-guide.md:21]:
```suggestion
| -s or --statement-size | Control the size of Insert Statement. Unit: byte. |
```

File: docs/en/user-guide.md [hunk 22]
```diff
@@ -0,0 +1,42 @@
...
+| -F or --filesize | The approximate size of output file, unit is byte. |
```
Comment (ran-huang) [inline docs/en/user-guide.md:22]:
```suggestion
| -F or --filesize | The approximate size of the output file. Unit: byte. |
```

File: docs/en/user-guide.md [hunk 23]
```diff
@@ -0,0 +1,42 @@
...
+| --filetype| The type of dump file. (sql/csv, default "sql")           |
+| -o or --output | Output directory. (We have a default value based on time.) |
```
Comment (ran-huang) [inline docs/en/user-guide.md:24]:
```suggestion
| -o or --output | Output directory. The default value is based on time. |
```

File: docs/en/user-guide.md [hunk 24]
```diff
@@ -0,0 +1,42 @@
...
+| --consistency | Which consistency control to use (default `auto`):<br>`flush`: Use FTWRL (flush tables with read lock)<br>`snapshot`: use a snapshot at given timestamp<br>`lock`: execute lock tables read for all tables need to be locked <br>`none`: dump without locking, cannot guarantee consistency <br>`auto`: `flush` on MySQL, `snapshot` on TiDB |
```
Comment (ran-huang) [inline docs/en/user-guide.md:25]:
```suggestion
| --consistency | Which consistency control to use (default `auto`):<br>`flush`: Use FTWRL (flush tables with read lock)<br>`snapshot`: use a snapshot at a given timestamp<br>`lock`: execute lock tables read for all tables that need to be locked <br>`none`: dump without locking. It cannot guarantee consistency <br>`auto`: `flush` on MySQL, `snapshot` on TiDB |
```

File: docs/en/user-guide.md [hunk 25]
```diff
@@ -0,0 +1,42 @@
...
+| --snapshot | Snapshot position. Valid only when consistency=snapshot. |
+| --where | Specify the dump range by where condition. Dump only selected records. |
```
Comment (ran-huang) [inline docs/en/user-guide.md:27]:
```suggestion
| --where | Specify the dump range by `where` condition. Dump only the selected records. |
```

File: docs/en/user-guide.md [hunk 26]
```diff
@@ -0,0 +1,42 @@
...
+| -p or --password | User password. |
+| -P or --port | TCP/IP port to connect to. (default 4000) |
```
Comment (ran-huang) [inline docs/en/user-guide.md:29]:
```suggestion
| -P or --port | TCP/IP port to connect to. (default: `4000`) |
```

File: docs/en/user-guide.md [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+| -u or --user | Username with privileges to run the dump. (default "root") |
+
+More detailed usage can be find by the flag `-h` or `--help`.
```
Comment (ran-huang) [inline docs/en/user-guide.md:32]:
```suggestion
To see more detailed usage, run the flag `-h` or `--help`.
```

File: docs/en/user-guide.md [hunk 28]
```diff
@@ -0,0 +1,42 @@
...
+
+## Mydumper Reference
+
+[Mydumper usage](https://github.com/maxbube/mydumper/blob/master/docs/mydumper_usage.rst)
+
+[TiDB Mydumper Manual](https://pingcap.com/docs/stable/reference/tools/mydumper/)
```
Comment (ran-huang) [inline docs/en/user-guide.md:38]:
```suggestion
[TiDB Mydumper Instructions](https://pingcap.com/docs/stable/reference/tools/mydumper/)
```

File: docs/en/user-guide.md [hunk 29]
```diff
@@ -0,0 +1,42 @@
...
+
+## Download
+
+[nightly](https://download.pingcap.org/dumpling-nightly-linux-amd64.tar.gz)
```
Comment (ran-huang) [inline docs/en/user-guide.md:42]:
```suggestion
Download Dumpling at [nightly](https://download.pingcap.org/dumpling-nightly-linux-amd64.tar.gz).
```

File: docs/en/user-guide.md (hunk 18 above)
Comment (YuJuncen) [reply docs/en/user-guide.md:15]: This description is from the CLI... I think this is more clear...

Comment (ran-huang) [reply docs/en/user-guide.md:15]: OK

File: docs/en/user-guide.md (hunk 29 above)
Comment (YuJuncen) [reply docs/en/user-guide.md:42]: How about `Download nightly version of Dumpling [here](download_url).`?

Comment (ran-huang) [reply docs/en/user-guide.md:42]: Don't forget `the` 😀
`Download the nightly version of Dumpling [here](download_url).`

Comment (YuJuncen) [reply docs/en/user-guide.md:42]: OK 😁

**Pull request merged**